app/controllers/grade_entry_forms_controller.rb

```diff
@@ -216,19 +216,15 @@ def download
   def upload
     @grade_entry_form = record
     begin
-      data = process_file_upload
-    rescue Psych::SyntaxError => e
-      flash_message(:error, t('upload_errors.syntax_error', error: e.to_s))
+      data = process_file_upload(['.csv'])
     rescue StandardError => e
       flash_message(:error, e.message)
     else
-      if data[:type] == '.csv'
-        overwrite = params[:overwrite]
-        grades_file = data[:file]
-        result = @grade_entry_form.from_csv(grades_file.read, overwrite)
-        flash_message(:error, result[:invalid_lines]) unless result[:invalid_lines].empty?
-        flash_message(:success, result[:valid_lines]) unless result[:valid_lines].empty?
-      end
+      overwrite = params[:overwrite]
+      grades_file = data[:file]
+      result = @grade_entry_form.from_csv(grades_file.read, overwrite)
+      flash_message(:error, result[:invalid_lines]) unless result[:invalid_lines].empty?
+      flash_message(:success, result[:valid_lines]) unless result[:valid_lines].empty?
```

Comment:
These two lines are duplicated across multiple CSV upload controller actions. This is a good opportunity to define a helper method like `flash_csv_result` and use that consistently throughout the various controllers. (This wasn't part of the original plan for this pull request, but because it's a relatively simple change and falls under the broader goal of "make the handling of controller upload actions more consistent", I'm happy to add it to this PR.)